libc/stdio_scanf.c

```diff
@@ -0,0 +1,3701 @@
+
+	TEST_ASSERT_EQUAL_INT(5, sscanf(tested_str, ext2, &max_float, &hal_float, &zero, &half_min_float, &min_float));
+	TestAssertFloat(max_float, hal_float, zero, half_min_float, min_float);
+}
+
+/*
+////////////////////////////////////////////////////////////////////////////////////////////////////
+*/
+
+TEST(stdio_scanf_float, inf_nan)
+{
+	char tested_str[400];
+	double inf, this_nan, neginf;
+	double expinf = HUGE_VALF;
+
+	char ext[] = "%lf %lf %lf";
+	sprintf(tested_str, ext, HUGE_VALF, HUGE_VALF * -1, NAN);
+	create_file(std_path, tested_str);
+	rd = fopen(std_path, "r");
+
+	TEST_ASSERT_EQUAL_INT(3, vfscand_wrapper(rd, ext, &inf, &neginf, &this_nan));
+
+	TEST_ASSERT_EQUAL_DOUBLE(expinf, inf);
+	TEST_ASSERT_EQUAL_DOUBLE(HUGE_VAL * -1, neginf);
+	TEST_ASSERT_EQUAL_DOUBLE(NAN, this_nan);
+
+	rewind(rd);
+
+	TEST_ASSERT_EQUAL_INT(3, fscanf(rd, ext, &inf, &neginf, &this_nan));
+
+	TEST_ASSERT_EQUAL_DOUBLE(expinf, inf);
+	TEST_ASSERT_EQUAL_DOUBLE(HUGE_VAL * -1, neginf);
+	TEST_ASSERT_EQUAL_DOUBLE(NAN, this_nan);
+
+	TEST_ASSERT_EQUAL_INT(
+		3, vsscanf_wrapper(tested_str, ext, &inf, &neginf, &this_nan));
+
+	TEST_ASSERT_EQUAL_DOUBLE(expinf, inf);
+	TEST_ASSERT_EQUAL_DOUBLE(HUGE_VAL * -1, neginf);
+	TEST_ASSERT_EQUAL_DOUBLE(NAN, this_nan);
+
+	TEST_ASSERT_EQUAL_INT(3, sscanf(tested_str, ext, &inf, &neginf, &this_nan));
+
+	TEST_ASSERT_EQUAL_DOUBLE(expinf, inf);
+	TEST_ASSERT_EQUAL_DOUBLE(HUGE_VAL * -1, neginf);
+	TEST_ASSERT_EQUAL_DOUBLE(NAN, this_nan);
+}
+
+/*
+////////////////////////////////////////////////////////////////////////////////////////////////////
+*/
+
+TEST_SETUP(stdio_scanf_str_chr_ptr) { }
+
+TEST_TEAR_DOWN(stdio_scanf_str_chr_ptr) { unlink(std_path); }
+
+TEST(stdio_scanf_str_chr_ptr, c)
+{
+
+	char tested_str[5], tested_str2[5];
+	char chr;
+	char T = 'T';
+	char tstr[] = "Test";
+
+	sprintf(tested_str, "%c", T);
+	create_file(std_path, tested_str);
+	rd = fopen(std_path, "r");
+
+	TEST_ASSERT_EQUAL_INT(1, vfscand_wrapper(rd, "%c", &chr));
+	TEST_ASSERT_EQUAL_CHAR(T, chr);
+
+	rewind(rd);
+
+	TEST_ASSERT_EQUAL_INT(1, fscanf(rd, "%c", &chr));
+	TEST_ASSERT_EQUAL_CHAR(T, chr);
+
+	TEST_ASSERT_EQUAL_INT(1, vsscanf_wrapper(tested_str, "%c", &chr));
+	TEST_ASSERT_EQUAL_CHAR(T, chr);
+
+	TEST_ASSERT_EQUAL_INT(1, sscanf(tested_str, "%c", &chr));
+	TEST_ASSERT_EQUAL_CHAR(T, chr);
+
+	sprintf(tested_str2, "%s", tstr);
+	create_file(std_path, tested_str2);
+	rd = fopen(std_path, "r");
+
+	TEST_ASSERT_EQUAL_INT(1, vfscand_wrapper(rd, "%c", &chr));
+	TEST_ASSERT_EQUAL_CHAR(tstr[0], chr);
+
+	rewind(rd);
+
+	TEST_ASSERT_EQUAL_INT(1, fscanf(rd, "%c", &chr));
+	TEST_ASSERT_EQUAL_CHAR(tstr[0], chr);
+
+	TEST_ASSERT_EQUAL_INT(1, vsscanf_wrapper(tested_str, "%c", &chr));
+	TEST_ASSERT_EQUAL_CHAR(tstr[0], chr);
+
+	TEST_ASSERT_EQUAL_INT(1, sscanf(tested_str, "%c", &chr));
+	TEST_ASSERT_EQUAL_CHAR(tstr[0], chr);
+}
+
+TEST(stdio_scanf_str_chr_ptr, str_path)
+{
+
+	char tested_str[50];
+	char chr[50];
```

Comment:
```suggestion
	char tested_str[50];
	char chr[50];
```

 - Please follow the variable naming convention
 - these names are a little misleading - both are strings, yet one is called chr (this applies in the next tests also)
 - magic sizes of buffers. what about a macro?